src/js/pages/NotificationSettings/index.jsx

```diff
@@ -0,0 +1,131 @@
+import React, {Component, Fragment} from "react";
+import {withNamespaces} from "react-i18next";
+import PropTypes from "prop-types";
+import {FormGroup, Label, Button, Alert} from 'reactstrap';
+import Input from 'react-validation/build/input';
+import Form from 'react-validation/build/form';
+import Switch from "react-switch";
+import {connect} from "react-redux";
+import emailNotifications from '../../features/emailNotifications';
+import {required, isEmail} from "../../validators";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faCircleNotch} from "@fortawesome/free-solid-svg-icons";
+
+class NotificationSettings extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showEmailSection: props.isSubscribed,
+      email: props.email,
+      emailValid: false
+    };
+    this.props.checkSubscription();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.isSubscribed !== this.props.isSubscribed && this.props.isSubscribed === true) {
+      this.setState({showEmailSection: this.props.isSubscribed});
+    }
+    // TODO find a way to retrieve the email
```

Comment:
This one might be tricky. We need a way to safely return the email from the service to avoid anyone being able to know the user's email address by requesting it to the API using only the address.

Maybe the user needs to sign a phrase, send it to the API and if the signature is valid the email is returned?

Another possibility is displaying in the notification settings that the user is subscribed, but not showing the email.

Reply:
Like you said, we could just not show the email field, it's really not the end of the world, it's a nice to have.
I guess we could do something similar to the unsubscribe, where we send the Dapp ID signed, with the account, so the API can know that it's us, but requiring the user to sign something just for a "get" sounds really bad.
Let's just leave it as a TODO for now